roles/contrail_node/tasks/main.yaml

```diff
@@ -0,0 +1,47 @@
+---
+- name: Untar Contrail tar-file
+  unarchive:
+    src: "{{ contrail_docker_images_path }}/contrail-kubernetes-docker-images_{{ contrail_version }}.tgz"
+    dest: "/tmp"
+  when: docker_registry is not defined
```

Comment:
Can you put all tasks with the `docker_registry is not defined` under a block with a single condition?